app/Http/Controllers/CampaignRegistrationController.php

```diff
@@ -53,4 +53,26 @@ public function store(Request $request)
 
         return redirect('campaign.index');
     }
+
+    public function destroy(Request $request, Campaign $campaign)
+    {
+        $data = $request->validate([
+            'user_id' => 'sometimes|array'
+        ]);
+
+        $user = Auth::user();
+        $isCampaignOwner = $campaign->users()->where('user_id', $user->id)->first()->pivot->is_owner;
+
+        if ($request->has('user_id') && $isCampaignOwner) {
+            $campaignCharacters = $campaign->characters()->whereIn('user_id', $data['user_id'])->get();
+
+            $campaign->users()->detach($data['user_id']);
+            $campaign->characters()->detach($campaignCharacters);
+
+            $campaign->code = $campaign->generateCode();
+            $campaign->save();
+        }
```

Comment:
pretty good, just missing some error handling here, in cases like this its better to exit early when the state lacks authorization: 
```suggestion
        $user = Auth::user();
        $isCampaignOwner = $campaign->users()->where('user_id', $user->id)->first()->pivot->is_owner;

        if (!$request->has('user_id')) {
            return redirect()->back()->withErrors(['error' => "You need to specify a user."])
        }
        
        if (!$isCampaignOwner) {
            return redirect()->back()->withErrors(['error' => "You don't have permission to do that."])
        }
        
        $campaignCharacters = $campaign->characters()->whereIn('user_id', $data['user_id'])->get();
        
        $campaign->users()->detach($data['user_id']);
        $campaign->characters()->detach($campaignCharacters);

        $campaign->code = $campaign->generateCode();
        $campaign->save();
        
```

Reply:
I agree, I added the changes!